src/asmcnc/apps/systemTools_app/screens/xy_jig/screen_xy_jig.py

```diff
@@ -0,0 +1,566 @@
+
+        self.load_home_peak.text = '-'
+        self.load_away_peak.text = '-'
+        self.load_home_average.text = '-'
+        self.load_away_average.text = '-'
+        self.load_graph_away.opacity = 0
+        self.load_graph_home.opacity = 0
+
+        self.m.is_machine_completed_the_initial_squaring_decision = True
+        self.m.is_squaring_XY_needed_after_homing = False
+        self.m.request_homing_procedure('xy_jig','xy_jig')
+
+    def on_enter(self):
+        if self.test_waiting_to_start:
+            self.test_waiting_to_start = False
+            Clock.schedule_once(self.begin_test, 1)
+
+    def begin_test(self, dt):
+        if self.test_running:
+            if self.m.state().startswith('Idle'):
+                self.m.jog_absolute_single_axis(self.axis, -1, self.max_speed)
+                Clock.schedule_once(self.start_moving_away, 1)
+            else:
+                Clock.schedule_once(self.begin_test, 0.1)
+
+    def start_moving_away(self, dt):
+        if self.test_running:
+            if self.m.state().startswith('Idle'):
+                self.m.jog_absolute_single_axis(self.axis, self.max_travel, self.max_speed / 5)
+                Clock.schedule_once(self.record_away_values, 0.4)
+            else:
+                Clock.schedule_once(self.start_moving_away, 0.1)
+
+    def record_away_values(self, dt):
+        if self.test_running:
+            if self.m.state().startswith('Idle'):
+                self.m.jog_absolute_single_axis(self.axis, -1, self.max_speed / 5)
+                Clock.schedule_once(self.record_home_values, 0.4)
+            else:
+                if self.axis == 'Y':
+                    sg_value = self.m.s.sg_y_axis
+                    sg_value_motor_1 = self.m.s.sg_y1_motor
+                    sg_value_motor_2 = self.m.s.sg_y2_motor
+                    pos = self.m.mpos_y()
+                else:
+                    sg_value = self.m.s.sg_x_motor_axis
+                    sg_value_motor_1 = self.m.s.sg_x1_motor
+                    sg_value_motor_2 = self.m.s.sg_x2_motor
+                    pos = self.m.mpos_x()
+
+                if sg_value not in [-999, None]:
+                    self.sg_values_away.append(sg_value)
+                    self.pos_values_away.append(pos)
+
+                if sg_value_motor_1 not in [-999, None]:
+                    self.sg_values_away_motor_1.append(sg_value_motor_1)
+                    self.pos_values_away_motor_1.append(pos)
+
+                if sg_value_motor_2 not in [-999, None]:
+                    self.sg_values_away_motor_2.append(sg_value_motor_2)
+                    self.pos_values_away_motor_2.append(pos)
+
+                Clock.schedule_once(self.record_away_values, 0.1)
+
+    def record_home_values(self, dt):
+        if self.test_running:
+            if self.m.state().startswith('Idle'):
+                PopupPhaseTwo(self.systemtools_sm.sm, self.l)
+            else:
+                if self.axis == 'Y':
```

Comment:
I remember choosing not to do this because it makes the logic quite confusing - fully functionalising would require either a function that takes 6 lists as arguments and returns another 6 which would need to each be reassigned to their respective variables, which seems a bit ugly to me, or something like 7 return values including the pos/sg values as well as booleans that represent whether these new values need adding to the lists (otherwise all the logic would just end up outside of the function anyway).

Or am I missing something obvious here? 😅